atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/KeyValueServiceDataTrackerTest.java

```diff
@@ -0,0 +1,278 @@
+        tracker.readForTable(TABLE_2, KVS_METHOD_NAME_2, BYTES_READ_2);
+        tracker.partialReadForTable(TABLE_1, BYTES_READ_3);
+        tracker.tableAgnosticRead(KVS_METHOD_NAME_5, BYTES_READ_2);
+        tracker.callForTable(TABLE_1);
+        tracker.partialReadForTable(TABLE_2, BYTES_READ_1);
+        tracker.callForTable(TABLE_3);
+        tracker.partialReadForTable(TABLE_2, BYTES_READ_2);
+        tracker.partialReadForTable(TABLE_3, BYTES_READ_2);
+        tracker.partialReadForTable(TABLE_1, BYTES_READ_2);
+        tracker.readForTable(TABLE_2, KVS_METHOD_NAME_3, BYTES_READ_3);
+        tracker.readForTable(TABLE_3, KVS_METHOD_NAME_1, BYTES_READ_2);
+        tracker.partialReadForTable(TABLE_3, BYTES_READ_1);
+        tracker.partialReadForTable(TABLE_3, BYTES_READ_1);
+
+        assertThat(tracker.getReadInfo())
+                .isEqualTo(createTransactionReadInfo(
+                        5 * BYTES_READ_1 + 6 * BYTES_READ_2 + 4 * BYTES_READ_3,
+                        12,
+                        createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_2)));
+
+        assertThat(tracker.getReadInfoByTable())
+                .containsExactlyEntriesOf(ImmutableMap.of(
+                        TABLE_1,
+                                createTransactionReadInfo(
+                                        BYTES_READ_1 + BYTES_READ_2 + 3 * BYTES_READ_3,
+                                        5,
+                                        createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_2)),
+                        TABLE_2,
+                                createTransactionReadInfo(
+                                        BYTES_READ_1 + 2 * BYTES_READ_2 + BYTES_READ_3,
+                                        3,
+                                        createMaximumBytesKvsCallInfo(BYTES_READ_3, KVS_METHOD_NAME_3)),
+                        TABLE_3,
+                                createTransactionReadInfo(
+                                        2 * BYTES_READ_1 + 2 * BYTES_READ_2,
+                                        2,
+                                        createMaximumBytesKvsCallInfo(BYTES_READ_2, KVS_METHOD_NAME_1))));
+    }
+
+    private static TransactionReadInfo createTransactionReadInfo(
+            long bytesRead, long kvsCalls, KvsCallReadInfo maximumBytesKvsCallInfo) {
+        return ImmutableTransactionReadInfo.builder()
+                .bytesRead(bytesRead)
+                .kvsCalls(kvsCalls)
+                .maximumBytesKvsCallInfo(maximumBytesKvsCallInfo)
+                .build();
+    }
+
+    private static KvsCallReadInfo createMaximumBytesKvsCallInfo(long bytesRead, String kvsMethodName) {
+        return ImmutableKvsCallReadInfo.builder()
+                .bytesRead(bytesRead)
+                .methodName(kvsMethodName)
+                .build();
+    }
+
+    private static TransactionReadInfo createTransactionReadInfo(long bytesRead, int kvsCalls) {
```

Comment:
nit: have overloads next to each other (shorter signature on top in this case, even though it's not delegating)